ceph/ceph.py

```diff
@@ -1913,6 +1916,655 @@ def enable_ceph_mgr_restful(self):
+                realm_zone = "realm.zone"
+
+                if realm_zone not in rgws:
+                    rgws[realm_zone] = []
+                id_ = "{}.{}.rgw".format(realm_zone, rgw.hostname)
+                rgws[realm_zone].append(
+                    "{host}={id}".format(host=rgw.hostname, id=id_)
+                )
+                daemons.append(id_)
+
+            # for realm_zone in rgws.keys():
+            #     (realm, zone) = realm_zone.split('.')[0:2]
+            #
+            #     radosgw_cmds = {
+            #         'realm': [
+            #             'radosgw-admin', 'realm', 'create',
+            #             '--rgw-realm', realm, '--default'
+            #         ],
+            #         'zone_group': [
+            #             'radosgw-admin', 'zonegroup', 'create',
+            #             '--rgw-zonegroup=default', '--master', '--default'
+            #         ],
+            #         'zone': [
+            #             'radosgw-admin', 'zone', 'create',
+            #             '--rgw-zonegroup=default', '--rgw-zone', zone,
+            #             '--master', '--default'
+            #         ]
+            #     }
+            #
+            #     for obj, cmd in radosgw_cmds.items():
+            #         try:
+            #             logger.info("creating {}...".format(obj))
+            #             self.shell(
+            #                 remote=self.installer,
+            #                 args=cmd,
+            #             )
+            #         except CommandFailed as warn:
+            #             logger.warning(warn)
+
+            for realm_zone, nodes in rgws.items():
+                (realm, zone) = realm_zone.split('.')
+                self.shell(
+                    remote=self.installer,
+                    args=[
+                        'ceph', 'orch', 'apply', 'rgw',
+                        realm, zone, '--placement',
+                        '"{};{}"'.format(str(len(nodes)), ';'.join(nodes))
+                    ]
+                )
+
+            # check daemon existence
+            assert self.check_exist(
+                daemon='rgw',
+                ids=daemons,
+                timeout=len(rgws) * self.TIMEOUT,
+            )
+
+        elif op == "purge":
+            # todo: Handle rgw purge
+            pass
+
+    # @staticmethod
+    # def test_disk(node, disk):
+    #     """
+    #     test disk
+    #     Args:
+    #         remote: remote object
+    #         disk: name of disk
+    #     Returns:
+    #         boolean
+    #     """
+    #     try:
+    #         node.exec_command(
+    #             cmd='stat {disk} && sudo dd if={disk} of=/dev/null count=1'
+    #                 ' && ! mount | grep -q {disk}'.format(disk=disk)
+    #         )
+    #         return True
+    #     except:
+    #         logger.debug("get_scratch_devices: %s is in use" % disk)
+    #     return False
```

Comment:
same as above, do we need this block of commented code.

Reply:
Initially added to check the disk. we can eliminate this part currently.